fix: add existing instance name variable

MatthewLemmond opened the pull request:
### Description
Update the module to no longer overload the var.key_protect_instance_name by using it as both the name for a resource that is provisioned or an existing instance name if not creating the instance, the existing instance functionality is now replaced with a variable for the existing instance name.

**Check the relevant boxes:**
- [ ] Bug fix (nonbreaking change that fixes an issue)
- [ ] New feature (nonbreaking change that adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Examples or tests (addition or updates of examples or tests)
- [ ] Documentation update
- [ ] CI related update (pipeline, etc.)

### Checklist

- [x] If relevant, a test for the change has been added or updated as part of this PR.
- [x] If relevant, documentation for the change has been added or updated as part of this PR.

### Merge

- Merge using "Squash and merge".
- Make sure to use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents. The commit message determines whether a new version of the modules needs to be released, and if so, which semver number to use).

ocofaigh reviewed (changes requested):
Instead of adding `var.existing_key_protect_instance_name` and doing a data lookup to get the ID, why not just add an input variable to pass the ID. This is also safer incase there are instances created with the same name. So the input var would be `var.var.existing_key_protect_instance_guid`

MatthewLemmond commented:
I have replaced the existing name with the existing GUID, since we cannot do a data lookup on the GUID we are unable to get the name for the existing instance and no modules using this module require the name to my knowledge so I have removed the name altogether (except for the input variable for controlling the initial name) as there was not a way to resolve the name for the existing instance to give to the output on [this line](https://github.com/terraform-ibm-modules/terraform-ibm-key-protect-all-inclusive/blob/main/main.tf#L8).

ocofaigh commented on variables.tf:
```diff
@@ -25,7 +25,13 @@ variable "enable_metrics" {
+
+variable "existing_key_protect_instance_guid" {
+  type        = string
+  description = "The GUID of an existing Key Protect instance, required if 'var.create_key_protect_instance' is false."
```
can we add variable validation in main.tf to ensure this is caught - aka if `key_protect_instance_name` is `null` then `existing_key_protect_instance_guid` cannot be `null` too

ocofaigh reviewed (commented):
@MatthewLemmond left a comment - nearly there

MatthewLemmond commented on outputs.tf:
```diff
@@ -7,11 +7,6 @@ output "key_protect_guid" {
-output "key_protect_name" {
-  description = "Name of the Key Protect instance"
-  value       = local.key_protect_name
-}
```
How about a hard-coded string like 
```hcl
local {
  key_protect_name = var.create_key_protect_instance ? module.key_protect[0].key_protect_name : "existing-instance"
}
```

ocofaigh replied:
No, just set the value to var.key_protect_name. That means if it is null, it wont output anything

ocofaigh commented on outputs.tf:
```diff
@@ -7,9 +7,12 @@ output "key_protect_guid" {
-  description = "Name of the Key Protect instance"
-  value       = local.key_protect_name
+  description = "Key Protect Name"
+  value       = var.key_protect_instance_name
```
as per call, use the resource output here, not input variable

MatthewLemmond replied:
After some testing, I realized that we cannot use the resource output for this as it is behind a count, so if we try to have the output be `module.key_protect[0].key_protect_name` it will give an error if the `create_key_protect_instance` flag is false. As this is the case the output is currently still set as `var.key_protect_instance_name`

ocofaigh replied:
We should be able to handle this with logic. Default to null if `create_key_protect_instance` is set to `false`. See example:
https://github.com/terraform-ibm-modules/terraform-ibm-observability-instances/blob/main/outputs.tf#L20

ocofaigh commented on variables.tf:
```diff
@@ -25,7 +25,13 @@ variable "enable_metrics" {
 variable "key_protect_instance_name" {
   type        = string
-  description = "The name of the existing Key Protect instance, if 'create_key_protect_instance' is true a new instance will be created with this name."
+  description = "The name of the Key Protect instance to provision."
```
The name to give the Key Protect instance that will be provisioned by the module. Not used if value is passed for var.existing_key_protect_instance_guid
